packages/react/src/TransNoContext.tsx

```diff
@@ -107,12 +108,12 @@ export function TransNoContext(
   const Component: React.ComponentType<TransRenderProps> =
     component || FallbackComponent
 
-  return React.createElement(Component, i18nProps, translation)
+  return <Component {...i18nProps}>{translation}</Component>
 }
 
 const RenderFragment = ({ children }: TransRenderProps) => {
   // cannot use React.Fragment directly because we're passing in props that it doesn't support
-  return <React.Fragment>{children}</React.Fragment>
+  return <>{children}</>
```

Comment:
`RenderFragment` could return just `children` and not add any Fragment around it, it seems unnecessary

Reply:
Why RenderFragment cannot be removed

```tsx
const RenderFragment = ({ children }: TransRenderProps) => {
  return <>{children}</>
}
```

`RenderFragment` is used as a fallback component in `TransNoContext`:

```tsx
const FallbackComponent = defaultComponent || RenderFragment
// ...
<FallbackComponent {...i18nProps}>{translation}</FallbackComponent>
```

Where `i18nProps` contains `{ id, message, translation, children }`.

It's safer to leave it for now.